src/Microsoft.Graph/Extensions/DateTimeTimeZoneExtensions.cs

```diff
@@ -22,26 +22,37 @@ public static class DateTimeTimeZoneExtensions
         /// <returns></returns>
         public static DateTime ToDateTime(this DateTimeTimeZone dateTimeTimeZone)
         {
-            DateTime dateTime = DateTime.ParseExact(dateTimeTimeZone.DateTime, DateTimeFormat, CultureInfo.InvariantCulture);
+            DateTime parsedDateTime = DateTime.ParseExact(dateTimeTimeZone.DateTime, DateTimeFormat, CultureInfo.InvariantCulture);
 
             // Now we need to determine which DateTimeKind to set based on the time zone specified in the input object.
             TimeZoneInfo timeZoneInfo = TimeZoneInfo.FindSystemTimeZoneById(dateTimeTimeZone.TimeZone);
 
             DateTimeKind kind;
-            if (timeZoneInfo.Id == TimeZoneInfo.Utc.Id)
+            if(timeZoneInfo.StandardName == TimeZoneInfo.Utc.StandardName)
```

Comment:
Any reason for this change, the Id should be unique. Yes?

Reply:
Yes, but I noticed that if you specify your timezone as Africa/Nairobi for example, the id for the fetched `TimeZoneInfo` will be 'Africa/Nairobi' while the that of the method `TimeZone.Local.Id` will be "". Which makes them unequal despite the fact that we want the passed timezone for "Africa/Nairobi" and that of "TimeZone.Local" to match given that am in Nairobi. The StandardName for the 2 though match and it informed me to use the `StandardName` as opposed to `Id` to get the scenario that we are aiming for above. see screenshots below for further details

Details of timezone info when I pass 'Africa/Nairobi' as my DateTimeTimeZone.TimeZone as below
![image](https://github.com/microsoftgraph/msgraph-sdk-dotnet/assets/11347476/db5c16f1-2695-491b-b900-6aff0b36ee92)

Details of TimeZone.Local
![image](https://github.com/microsoftgraph/msgraph-sdk-dotnet/assets/11347476/fffb0213-40d2-4bbc-b79c-827fdd900a55)

as seen above, the 2 screenshots above represent utc+03:00 but have different Ids. What they share in common is the StandardName.

Reply:
Thanks for clarifying this.